### PR TITLE
[BOLT][DWARF][NFC] Add tests with multiple CUs

### PR DESCRIPTION
**Pull request by sayhaan**
Adds DWARF4 and DWARF5 tests with multiple CUs.

### CODE REVIEW

File: bolt/test/X86/dwarf4-subprogram-multiple-ranges-cus.test [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -dwarf-version=4 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf4-subprogram-multiple-ranges-main.s -o %t1.o
+# RUN: llvm-mc -dwarf-version=4 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf4-subprogram-multiple-ranges-other.s -o %t1.o
```
Comment (ayermolo) [inline bolt/test/X86/dwarf4-subprogram-multiple-ranges-cus.test:4]: You are overriding object file.
Line below should have two object files as input.
Once that is fixed, you should see linker error, because your new inputs have exact the same function names.

File: bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-subprogram-multiple-ranges-main.s -o %t1.o
+# RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-subprogram-multiple-ranges-other.s -o %t1.o
```
Comment (ayermolo) [inline bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test:4]: Same here.

File: bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# This test checks BOLT correctly handles DW_TAG_subprogram with Ranges with multiple entries.
+
+# POSTCHECK: _Z7doStuffi>:
+# POSTCHECK: [[#%.6x,ADDR:]]
+# POSTCHECK: _Z7doStuffi.__part.1>:
+# POSTCHECK-NEXT: [[#%.6x,ADDR1:]]
+# POSTCHECK: _Z7doStuffi.__part.2>:
+# POSTCHECK-NEXT: [[#%.6x,ADDR2:]]
+
+# POSTCHECK: DW_TAG_subprogram
```
Comment (ayermolo) [inline bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test:20]: In both tests you should check ranges in both CUs.

File: bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-subprogram-multiple-ranges-main.s -o %t1.o
+# RUN: llvm-mc -dwarf-version=5 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf5-subprogram-multiple-ranges-other.s -o %t2.o
+# RUN: %clang %cflags %t1.o %t2.o -o %t.exe -Wl,-q
+# RUN: llvm-bolt %t.exe -o %t.bolt --update-debug-sections
+# RUN: llvm-objdump %t.bolt --disassemble > %t1.txt
+# RUN: llvm-dwarfdump --show-form --verbose --debug-info %t.bolt >> %t1.txt
+# RUN: cat %t1.txt | FileCheck --check-prefix=POSTCHECK %s
+
+# This test checks BOLT correctly handles DW_TAG_subprogram with Ranges with multiple entries and multiple CUs.
```
Comment (ayermolo) [inline bolt/test/X86/dwarf5-subprogram-multiple-ranges-cus.test:11]: double hashes
"This test checks that BOLT"
later part maybe rephrase something like "handles multiple CUs with ranges".

File: bolt/test/X86/dwarf4-subprogram-multiple-ranges-cus.test [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# RUN: llvm-mc -dwarf-version=4 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf4-subprogram-multiple-ranges-main.s -o %t1.o
+# RUN: llvm-mc -dwarf-version=4 -filetype=obj -triple x86_64-unknown-linux %p/Inputs/dwarf4-subprogram-multiple-ranges-other.s -o %t2.o
+# RUN: %clang %cflags %t1.o %t2.o -o %t.exe -Wl,-q
+# RUN: llvm-bolt %t.exe -o %t.bolt --update-debug-sections
+# RUN: llvm-objdump %t.bolt --disassemble > %t1.txt
+# RUN: llvm-dwarfdump --show-form --verbose --debug-info %t.bolt >> %t1.txt
+# RUN: cat %t1.txt | FileCheck --check-prefix=POSTCHECK %s
+
+# This test checks BOLT correctly handles DW_TAG_subprogram with Ranges with multiple entries and multiple CUs.
```
Comment (ayermolo) [inline bolt/test/X86/dwarf4-subprogram-multiple-ranges-cus.test:11]: same here

File: bolt/test/X86/Inputs/dwarf5-subprogram-multiple-ranges-other.s [hunk 6]
```diff
@@ -0,0 +1,385 @@
+# clang++ -fbasic-block-sections=all -ffunction-sections -g2 -gdwarf-5
+# int doStuffOther(int val) {
+#   if (val)
+#     ++val;
+#   return val;
+# }
+#
+# int mainOther(int argc, const char** argv) {
```
Comment (ayermolo) [inline bolt/test/X86/Inputs/dwarf5-subprogram-multiple-ranges-other.s:8]: Are you sure your cpp actually contained mainOther?

File: bolt/test/X86/Inputs/dwarf4-subprogram-multiple-ranges-other.s [hunk 7]
```diff
@@ -0,0 +1,331 @@
+# clang++ -fbasic-block-sections=all -ffunction-sections -g2 -gdwarf-4
+# int doStuffOther(int val) {
+#   if (val)
+#     ++val;
+#   return val;
+# }
+#
+# int mainOther(int argc, const char** argv) {
```
Comment (ayermolo) [inline bolt/test/X86/Inputs/dwarf4-subprogram-multiple-ranges-other.s:8]: Same here

File: bolt/test/X86/Inputs/dwarf5-subprogram-multiple-ranges-other.s (hunk 6 above)
Comment (sayhaan) [reply bolt/test/X86/Inputs/dwarf5-subprogram-multiple-ranges-other.s:8]: Regenerated .s to make sure it's the correct ASM.

**Review by ayermolo — APPROVED** [review]
